src/js/components/lazy-load.js

```diff
@@ -11,49 +9,116 @@ const DEFAULTS = {
 }
 
 class LazyLoad {
+  /**
+   * Create a lazy load component
+   * Set base methods references and props (element and options)
+   *
+   * @param  {[DOM]} element
+   * @param  {[Object]} options
+   */
   constructor (element, options) {
-    this.$element = $(element)
-    this.options = $.extend({}, DEFAULTS, (options || {}))
+    this.element = element
+    this.options = { ...DEFAULTS, ...options }
+
+    this.onScrollHandler = this.onScrollHandler.bind(this)
+    this.onScroll = this.onScroll.bind(this)
+    this.checkPlaceholder = this.checkPlaceholder.bind(this)
   }
 
+  /**
+   * Set event listeners and check placeholder visibility
+   *
+   * @return {[Class]}
+   */
   init () {
     this.bindListeners()
     this.checkVisiblePlaceholders()
 
     return this
   }
 
+  /**
+   * Add scroll event to window and attach @onScrollHandler to event
+   */
   bindListeners () {
-    this.onScrollHandler = throttle(this.onScroll.bind(this), this.options.throttle)
-    $(window).on('scroll', this.onScrollHandler)
+    window.addEventListener('scroll', this.onScrollHandler)
+  }
+
+  /**
+   * Prevent @onScroll to be called multiple times using throttle
+   */
+  onScrollHandler () {
+    throttle(this.onScroll, this.options.throttle)
   }
 
+  /**
+   * Check if exist at least on [data-lazy] and check visibility
```

Comment:
`Check placeholders visibility if there is at least one registered in the element instance. If there are not placeholders to be checked, then the scroll event handler will be removed.`